### PR TITLE
Remove buildtools

### PR DESCRIPTION
**Pull request by sbomer**
I think this is finally ready to go. @dotnet/coreclr-infra this removes the dependency on buildtools. Please let me know if you think there's any additional testing I need to do.
- Makes a number of common test projects SDK-style, and replaces the buildtools prerelease restore target with the restore logic built in to the SDK. For some projects, we currently restore 5.0 assets that are used for netcoreapp3.0 - this required a workaround to switch the TFM in a few cases.
- Replaces the corefx testhost deps file creation logic with that from the SDK's publish logic.
- Uses a new IL sdk (as the old one used to make many redundant file copies), and uses an .ilproj to restore ilasm/ildasm up-front.
- Uses the UpdateVersions logic from arcade instead of buildtools.

This opens up the possibility of further simplifying some of our msbuild imports, but I don't want to do so in this PR as it's already a sizeable change.

### CODE REVIEW

File: dependencies.props [hunk 1]
```diff
@@ -17,7 +17,6 @@
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>
     <CoreClrCurrentRef>5d3c9a7c54c1c59b764de0e2dfb6bbb4ce29476c</CoreClrCurrentRef>
```
Comment (hoyosjs) [inline dependencies.props]: This was mostly used to update ILAsmVersion.txt. I see you are removing that file, should we just remove the MicrosoftNETCoreRuntimeCoreCLRPackageVersion update and make a DarcMaestro subscription for the CoreCLR packages? Also, if that happens, should we delete this file and move anything interesting left to Directory.Build.props? Are there scenarios that depend on this as it currently is?

Comment (sbomer) [reply dependencies.props]: Eventual goal is to get rid of this file - I'll look into it and may work on that in this change if it makes sense.

**Review by jashook — APPROVED** [review]
lgtm thank you for this change!

File: eng/updateversions.proj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<Project>
+
+  <PropertyGroup>
+    <ArcadeBuild>true</ArcadeBuild>
+  </PropertyGroup>
+
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.versiontools.tasks\$(MicrosoftDotNetVersionToolsTasksPackageVersion)\build\Microsoft.DotNet.VersionTools.Tasks.targets" />
```
Comment (hoyosjs) [inline eng/updateversions.proj]: Technically this is the only thing that we still have from buildtools?

Comment (sbomer) [reply eng/updateversions.proj]: This now comes from arcade: https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.VersionTools :)

File: tests/dir.props [hunk 3]
```diff
@@ -59,14 +31,6 @@
     <AltJitArch>$(__AltJitArch)</AltJitArch>
```
Comment (echesakovMSFT) [inline tests/dir.props]: We can git of this in the future.

File: tests/src/Common/Directory.Build.targets [hunk 4]
```diff
@@ -1,8 +1,39 @@
...
+    <ItemGroup>
+      <RunTimeDependecyExclude Include="$(CORE_ROOT)\**\*.*"  />
+      <RunTimeDependecyExcludeFiles Include="@(RunTimeDependecyExclude -> '%(FileName)%(Extension)')" />
+      <RunTimeDependecyExcludeFiles Include="@(RunTimeDependecyExclude -> '%(FileName).ni%(Extension)')" />
+      <RunTimeDependecyExcludeFiles Include="@(RunTimeDependecyExclude -> '%(FileName).pdb')" />
+      <AllResolvedRuntimeDependencies Include="@(RuntimeCopyLocalItems -> '%(FileName)%(Extension)');@(NativeCopyLocalItems -> '%(FileName)%(Extension)')">
+        <File>%(Identity)</File>
+      </AllResolvedRuntimeDependencies>
+      <RunTimeDependecyCopyLocalFile Include="@(AllResolvedRuntimeDependencies)"  Exclude="@(RunTimeDependecyExcludeFiles)"/>
+      <RunTimeDependecyCopyLocal Include="@(RunTimeDependecyCopyLocalFile -> '%(File)')"  />
```
Comment (echesakovMSFT) [inline tests/src/Common/Directory.Build.targets]: We can fix the typo in the future.

File: tests/src/Common/Directory.Build.targets [hunk 5]
```diff
@@ -1,8 +1,39 @@
...
+      <RunTimeDependecyCopyLocal Include="$(TargetingPackPath)/*" />
+    </ItemGroup>
+
+    <Copy
+      SourceFiles="@(RunTimeDependecyCopyLocal)"
```
Comment (echesakovMSFT) [inline tests/src/Common/Directory.Build.targets]: Typo

**Review by echesakov — APPROVED** [review]
Looks Great! Thank you!

**Pull request merged**